packages/devkit/src/generators/plugin-migrations/plugin-migration-utils.ts

```diff
@@ -0,0 +1,521 @@
+    delete target.executor;
+
+    deleteMatchingProperties(target, createdTarget);
+
+    target = postTargetTransformer(target, tree, projectFromGraph.data.root);
+
+    if (Object.keys(target).length > 0) {
+      projectConfig.targets[targetName] = target;
+    } else {
+      delete projectConfig.targets[targetName];
+    }
+    updateProjectConfiguration(tree, projectName, projectConfig);
+  }
+
+  const allConfigFilesAreIncluded = configFiles.every((configFile) => {
+    for (const includePattern of include) {
+      if (minimatch(configFile, includePattern, { dot: true })) {
+        return true;
+      }
+    }
+    return false;
+  });
+
+  if (allConfigFilesAreIncluded) {
+    include = undefined;
+  }
+
+  return { targetName, include, migratedProjects };
+}
+
+/**
+ * Add the plugin to nx.json using the specified target names
+ *
+ * If the plugin does not exist, add it with the specified target names
+ * If the plugin exists and the target names match the existing plugin, do nothing
+ * If the plugin exists and the target names do not match, add a new plugin entry scoped to just the projects listed in the include paths
+ *
+ * @example
+ * // Will set `test` and `test-ci` as the targetNames
+ *
+ * const include = ["myapp\/**\/*"];
+ * addPluginWithOptions<PlaywrightPluginOptions>(
+ *    tree,
+ *    '@nx/playwright/plugin',
+ *    include
+ *    { targetName: 'test', ciTargetName: 'test-ci' },
+ * );
+ *
+ * @param tree Virtual Tree
+ * @param pluginPath The name of the plugin to add to NxJson if it does not exist
+ * @param include Optional - An array of project paths that this plugin should only apply to
+ * @param pluginOptions The record of the names for the targets
+ */
+export function addPluginWithOptions<T extends object>(
+  tree: Tree,
+  pluginPath: string,
+  include: string[] | undefined,
+  pluginOptions: T
+): void {
+  const nxJson = readNxJson(tree);
+  nxJson.plugins ??= [];
+
+  const pluginExists = nxJson.plugins.find(
+    (plugin: ExpandedPluginConfiguration<T>) => {
+      if (typeof plugin === 'string' || plugin.plugin !== pluginPath) {
+        return;
+      }
+      for (const key in plugin.options) {
+        if (plugin.options[key] !== pluginOptions[key]) {
+          return false;
+        }
+      }
+
+      let includesContainsAllNewIncludePaths = !(!include && plugin.include);
+
+      if (include && plugin.include) {
+        for (const pluginIncludes of plugin.include) {
+          for (const projectPath of include) {
+            if (!minimatch(projectPath, pluginIncludes, { dot: true })) {
+              includesContainsAllNewIncludePaths = false;
+            }
+          }
+        }
+      }
+
+      return includesContainsAllNewIncludePaths;
+    }
+  );
+  if (pluginExists) {
+    return;
+  }
+
+  nxJson.plugins.push({
+    plugin: pluginPath,
+    include,
+    options: pluginOptions,
+  });
+
+  updateNxJson(tree, nxJson);
```

Comment:
Let's change this to update the existing plugin to update includes.

At the end check again if include includes every configfile... if it does... we can then remove the include property.

We will need update the playwright init.